cmd/mdatagen/loader.go

```diff
@@ -89,6 +93,10 @@ func (mvt ValueType) Primitive() string {
 		return "bool"
 	case pcommon.ValueTypeBytes:
 		return "[]byte"
+	case pcommon.ValueTypeSlice:
+		return "[]string"
+	case pcommon.ValueTypeMap:
+		return "map[string]string"
```

Comment:
I don't think we should restrict the elements type
```suggestion
	case pcommon.ValueTypeSlice:
		return "[]any"
	case pcommon.ValueTypeMap:
		return "map[string]any"
```

Reply:
`any` seems to wide open, which is why I had done it this way.  I originally had it as `any` so I can just revert the second commit on this PR.